File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let store = new Set(nums);
        let res = 0;
        for (let num of store) {
            let count = 0;
            if (!store.has(num-1)) {
                while (store.has(num+count)) {
                    count += 1;
                }
                res = Math.max(res,count);
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        res = 0
        l, r = 0, len(heights)-1
        while l < r:
            lower = min(heights[l],heights[r])
            res = max(res,lower*(r-l))
            if heights[l] == lower:
                l += 1
            else:
                r -= 1
        return res